hydrator-common/src/main/java/io/cdap/plugin/common/db/AbstractDBConnector.java

```diff
@@ -253,15 +270,51 @@ protected BrowseDetail listTables(Connection connection, @Nullable String schema
     return browseDetailBuilder.setTotalCount(count).build();
   }
 
-  protected Connection getConnection() {
+  private void validateDatabase(String database, Connection connection) throws SQLException {
+    if (database == null) {
+      return;
+    }
+    try (ResultSet catalogs = queryDatabases(connection)) {
+      boolean exits = false;
+      while (catalogs.next()) {
+        if (database.equals(catalogs.getString(RESULTSET_COLUMN_TABLE_CAT))) {
+          exits = true;
+          break;
+        }
+      }
+      if (!exits) {
+        throw new IllegalArgumentException(String.format("Database %s does not exist.", database));
+      }
+    }
+  }
+
+  protected void validateSchema(String database, String schema, Connection connection)
+    throws SQLException {
+    if (schema == null) {
+      return;
+    }
+    // NOTE Oracle schema name is case sensitive here
+    try (ResultSet schemas = connection.getMetaData()
+      .getSchemas(database == null ? connection.getCatalog() : database, schema)) {
+      if (!schemas.next()) {
+        throw new IllegalArgumentException(String.format("Schema '%s' does not exist.", schema));
+      }
+    }
+  }
+
+  protected Connection getConnection(@Nullable DBConnectorPath path) {
+    return getConnection(config.getConnectionString(), config.getConnectionArgumentsProperties());
+  }
+
+  protected Connection getConnection(String connectionString, Properties connectionArguments) {
```

Comment:
Do we still need this method if we have `getConnection(DBConnectorPath)`?

Reply:
this helper method can be reused in DB Specific plugin